### ISSUE TITLE
Missing parens in a method call is not detected

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#2658]
### Steps to reproduce
1. file
```python
def f():
    return False

if f:    # ERROR -- should be f()
    print("yes")

class C:
    def m(self):
        return False

if C().m:     # ERROR -- should be C().m()
    print("yes")
```

2. `epylint`

### Current behavior

`if f` (instead of `if f()`) is flagged:
```
 warning (W0125, using-constant-test, ) Using a conditional statement with a constant value
```
(apparently, per https://github.com/PyCQA/pylint/issues/2082)

### Expected behavior

Additionally, I expected `if C().m` (instead of `if C().m()`) to be flagged.

### pylint --version output

```
pylint 2.2.2
astroid 2.1.0
Python 3.7.1 (default, Nov  6 2018, 18:45:35) 
[Clang 10.0.0 (clang-1000.11.45.5)]
```

**Comment by PCManticore** [comment]
Thanks, that could definitely be improved.
